drivers/ma_yya_report/app/models/ma_yya_report/universe_calculator.rb

```diff
@@ -0,0 +1,206 @@
+###
+# Copyright 2016 - 2022 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+module MaYyaReport
+  class UniverseCalculator
+    include Filter::FilterScopes
+
+    def initialize(filter)
+      @filter = filter
+    end
+
+    def client_ids
+      client_scope.distinct.pluck(:id)
+    end
+
+    def calculate(&post_processor)
+      client_scope.find_in_batches do |batch|
+        post_processor.call(for_batch(batch))
+      end
+    end
+
+    private def for_batch(batch)
+      enrollments_by_client_id = clients_with_enrollments(batch)
+      {}.tap do |clients|
+        batch.each do |client|
+          client_id = client.id
+          enrollment = enrollments_by_client_id[client_id].last
```

Comment:
What's the chance we won't find a client in enrollments_by_client_id`  Might we want to next if enrollment.blank?

Reply:
I don't _think_ it will happen, but no harm in guarding against it -- especially if we turn out to need some of the filtering that some of the HUD reports need.